### PR TITLE
[Flight Fixture] Show SSR Support with CSS

### PR DESCRIPTION
**Pull request by sebmarkbage**
Builds on #26257.

To do this we need access to a manifest for which scripts and CSS are used for each "page" (entrypoint).

The initial script to bootstrap the app is inserted with `bootstrapScripts`. Subsequent content are loaded using the chunks mechanism built-in.

The stylesheets for each pages are prepended to each RSC payload and rendered using Float. This doesn't yet support styles imported in components that are also SSR:ed nor imported through Server Components. That's more complex and not implemented in the node loader.

HMR doesn't work after reloads right now because the SSR renderer isn't hot reloaded because there's no idiomatic way to hot reload ESM modules in Node.js yet. Without killing the HMR server. This leads to hydration mismatches when reloading the page after a hot reload.

Notably this doesn't show serializing the stream through the HTML like real implementations do. This will lead to possible hydration mismatches based on the data. However, manually serializing the stream as a string isn't exactly correct due to binary data. It's not the idiomatic way this is supposed to work. This will all be built-in which will make this automatic in the future.

### CODE REVIEW

File: fixtures/flight/server/global.js [hunk 1]
```diff
@@ -85,52 +108,64 @@ app.all('/', async function (req, res, next) {
...
+        buildPath = path.join(__dirname, '../build/');
+      }
+      // Read the module map from the virtual file system.
+      const moduleMap = JSON.parse(
+        await virtualFs.readFile(
+          path.join(buildPath, 'react-ssr-manifest.json'),
+          'utf8'
+        )
+      );
+      // Read the initial
```
Comment (gaearon) [inline fixtures/flight/server/global.js:132]: The initial what?

File: fixtures/flight/server/region.js [hunk 2]
```diff
@@ -45,25 +50,50 @@ app.get('/', async function (req, res) {
...
+  const root = [
+    // Prepend the App's tree with stylesheets required for this entrypoint.
+    mainCSSChunks.map(filename =>
+      React.createElement('link', {
+        rel: 'stylesheet',
+        href: filename,
+        precedence: 'default',
+      })
+    ),
+    React.createElement(App),
```
Comment (unstubbable) [inline fixtures/flight/server/region.js:91]: Why couldn't the `mainCSSChunks` be passed as prop to `App` which would then render the link elements in the head, similar to what is documented at https://beta.reactjs.org/reference/react-dom/server/renderToPipeableStream#reading-css-and-js-asset-paths-from-the-build-output?

Comment (sebmarkbage) [reply fixtures/flight/server/region.js:91]: It couldn't but this shows that you don't have to - using a new feature that places it in the right place. So the App code can't mess it up by forgetting.

Comment (unstubbable) [reply fixtures/flight/server/region.js:91]: Now I realise that this was a stupid question. I wasn't aware that this avoids the need to serialise them manually with `bootstrapScriptContent`, and instead "Float" is handling these elements automatically. Sorry for the noise my PR comments might create.

File: fixtures/flight/loader/global.js [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+      };
+    }
+    return {source: newResult.code};
+  }
+  return defaultTransformSource(source, context, defaultTransformSource);
+}
+
+export const transformSource =
+  process.version < 'v16' ? babelTransformSource : undefined;
+export const getSource = process.version < 'v16' ? getSourceImpl : undefined;
```
Comment (gnoff) [inline fixtures/flight/loader/global.js:52]: getting a `ReferenceError: getSourceImpl is not defined` error on node 14. Is this an oversight?

**Review by gnoff — APPROVED** [review]
Aside from the error on old node, I think this looks good

File: fixtures/flight/loader/global.js (hunk 3 above)
Comment (sebmarkbage) [reply fixtures/flight/loader/global.js]: Fixed. It still doesn't work on 14 though because undici isn't supported.

**Pull request merged**